src/PlusDataCollection/OptiTrack/vtkPlusOptiTrack.cxx

```diff
@@ -313,21 +331,32 @@ PlusStatus vtkPlusOptiTrack::InternalCallback(sFrameOfMocapData* data)
 {
   LOG_TRACE("vtkPlusOptiTrack::InternalCallback");
 
-  this->Internal->MatchTrackedTools();
-
   const double unfilteredTimestamp = vtkPlusAccurateTimer::GetSystemTime();
 
-  sDataDescriptions* dataDescriptions;
-  this->Internal->NNClient->GetDataDescriptions(&dataDescriptions);
-
-  int numberOfRigidBodies = data->nRigidBodies;
-  sRigidBodyData* rigidBodies = data->RigidBodies;
+  if (this->FrameNumber == 0)
+  {
+    // do an initial match of tracked tools
+    this->Internal->MatchTrackedTools();
+    this->Internal->LastToolUpdateTime = unfilteredTimestamp;
+  }
 
-  if (data->nOtherMarkers)
+  //this->Internal->MatchTrackedTools();
+  double timeSinceToolUpdate = unfilteredTimestamp - this->Internal->LastToolUpdateTime;
+  if (this->Internal->UpdateTools && timeSinceToolUpdate > this->Internal->ToolUpdateRate)
   {
-    LOG_WARNING("vtkPlusOptiTrack::InternalCallback: Untracked markers detected. Check for interference or make sure that the entire tool is in view")
+    this->Internal->MatchTrackedTools();
+    this->Internal->LastToolUpdateTime = unfilteredTimestamp;
   }
 
+  int numberOfRigidBodies = data->nRigidBodies;
+  sRigidBodyData* rigidBodies = data->RigidBodies;
+  
+  //TODO: Provide useful output to the user if untracked markers are found in the view, instead of this log spam
+  //if (data->nOtherMarkers)
+  //{
+  //  LOG_WARNING("vtkPlusOptiTrack::InternalCallback: Untracked markers detected. Check for interference or make sure that the entire tool is in view")
```

Comment:
You can remove this section. Or, if you think it could be useful then make the warning optional (controlled using a class member variable; maybe even make it adjustable from the config file).